src/phoenix/db/migrations/versions/deb2c81c0bb2_dataset_splits.py

```diff
@@ -0,0 +1,130 @@
+    # See https://docs.sqlalchemy.org/en/20/core/custom_types.html
+    __visit_name__ = "JSONB"
+
+
+@compiles(JSONB, "sqlite")
+def _(*args: Any, **kwargs: Any) -> str:
+    # See https://docs.sqlalchemy.org/en/20/core/custom_types.html
+    return "JSONB"
+
+
+JSON_ = (
+    JSON()
+    .with_variant(
+        postgresql.JSONB(),
+        "postgresql",
+    )
+    .with_variant(
+        JSONB(),
+        "sqlite",
+    )
+)
+
+_Integer = sa.Integer().with_variant(
+    sa.BigInteger(),
+    "postgresql",
+)
+
+
+# revision identifiers, used by Alembic.
+revision: str = "deb2c81c0bb2"
+down_revision: Union[str, None] = "58228d933c91"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    # Create dataset_splits table
+    op.create_table(
+        "dataset_splits",
+        sa.Column("id", _Integer, primary_key=True),
+        sa.Column("name", sa.String, nullable=False, unique=True),
+        sa.Column("description", sa.String, nullable=True),
+        sa.Column("color", sa.String, nullable=False),
+        sa.Column("metadata", JSON_, nullable=False),
+        sa.Column(
+            "created_at",
+            sa.TIMESTAMP(timezone=True),
+            nullable=False,
+            server_default=sa.func.now(),
+        ),
+        sa.Column(
+            "updated_at",
+            sa.TIMESTAMP(timezone=True),
+            nullable=False,
+            server_default=sa.func.now(),
+            onupdate=sa.func.now(),
+        ),
+    )
+
+    # Create crosswalk table: dataset_splits_dataset_examples
+    op.create_table(
+        "dataset_splits_dataset_examples",
+        sa.Column("id", _Integer, primary_key=True),
+        sa.Column(
+            "dataset_split_id",
+            _Integer,
+            sa.ForeignKey("dataset_splits.id", ondelete="CASCADE"),
+            nullable=False,
+        ),
+        sa.Column(
+            "dataset_example_id",
+            _Integer,
+            sa.ForeignKey("dataset_examples.id", ondelete="CASCADE"),
+            nullable=False,
+        ),
+        sa.UniqueConstraint(
+            "dataset_split_id",
+            "dataset_example_id",
+        ),
+    )
+
+    # Create experiments_dataset_splits table The rational of this table is to
+    # gather examples for a specific dataset split for a specific experiment.
+    # Select all dataset examples where examples belong to a dataset split and
+    # examples belong to a experiment.
+
+    op.create_table(
+        "experiments_dataset_splits",
+        sa.Column("id", _Integer, primary_key=True),
+        sa.Column(
+            "experiment_id",
+            _Integer,
+            sa.ForeignKey("experiments.id", ondelete="CASCADE"),
+            nullable=False,
+        ),
+        sa.Column(
+            "dataset_split_id",
+            _Integer,
+            sa.ForeignKey("dataset_splits.id", ondelete="CASCADE"),
+            nullable=False,
+        ),
```

Comment:
the index on the second element of the unique constraint tuple is still needed, because only the first element is sorted in the composite index behind the UNIQUE CONSTRAINT; the second element is not sorted.
```suggestion
            nullable=False,
            index=True,
        ),
```